include/triggeralgs/TriggerPrimitive.hpp

```diff
@@ -9,22 +9,92 @@
 #ifndef TRIGGERALGS_INCLUDE_TRIGGERALGS_TRIGGERPRIMITIVE_HPP_
 #define TRIGGERALGS_INCLUDE_TRIGGERALGS_TRIGGERPRIMITIVE_HPP_
 
+#include "triggeralgs/Types.hpp"
+
+#include <cstddef>
 #include <cstdint>
+#include <bitset>
 
 namespace triggeralgs {
+
+/**
+ * @brief A single energy deposition on a TPC or PDS channel
+ */
 struct TriggerPrimitive
 {
-  int64_t time_start = { 0 };
-  int64_t time_peak = { 0 };
-  int32_t time_over_threshold = { 0 };
-  uint16_t channel = { 0 };      // NOLINT(build/unsigned)
+  /**
+   * @brief The type of a TriggerPrimitive
+   */
+  enum class Type {
+    kUnknown = 0,
+    kTPC = 1,
+    kPDS = 2,
+  };
+
+  /**
+   * @brief The algorithm used to form a TriggerPrimitive
+   */
+  enum class Algorithm {
+    kUnknown = 0,
+    kTPCDefault = 1
+  };
+
+  /**
+   * Names for each of the bits in the TriggerPrimitive Flags
+   */
+  enum class FlagBits {
+    kSomehowBad = 0,
+    kUnassigned1 = 1,
+    kUnassigned2 = 2,
+    kUnassigned3 = 3,
+    kUnassigned4 = 4,
+    kUnassigned5 = 5,
+    kUnassigned6 = 6,
+    kUnassigned7 = 7,
+    kUnassigned8 = 8,
+    kUnassigned9 = 9,
+    kUnassigned10 = 10,
+    kUnassigned11 = 11,
+    kUnassigned12 = 12,
+    kUnassigned13 = 13,
+    kUnassigned14 = 14,
+    kUnassigned15 = 15,
+  };
```

Comment:
Consider defining these as powers of two instead, to avoid having to use `1u << static_cast<Flags>(b)` which could just be `static_cast<Flags>(b)`. Perhaps also include an operator to allow `|` and to generate composite masks like `kSomehowBad | kUnassigned7`
```
inline Flags operator|(FlagBits a, FlagBits b) {
    return static_cast<Flags>(a) | static_cast<Flags>(b);
}
inline Flags operator|(Flags a, FlagBits b) {
    return a | static_cast<Flags>(b);
}
inline Flags operator|(FlagBits a, Flags b) {
    return b | static_cast<Flags>(a);
}
```
`constexpr Flags` in a namespace may make more sense than an `enum class` here, since these are already implemented for `uint16_t`.